src/solvers/dgsem_tree/dg_2d.jl

```diff
@@ -490,6 +512,179 @@ end
 end
 
 
+function calc_volume_integral!(du, u,
+                               mesh::TreeMesh{2},
+                               nonconservative_terms, equations,
+                               volume_integral::VolumeIntegralShockCapturingSubcell,
+                               dg::DGSEM, cache)
+  @unpack indicator = volume_integral
+
+  @threaded for element in eachelement(dg, cache)
+    subcell_limiting_kernel!(du, u, element, mesh,
+                             nonconservative_terms, equations,
+                             volume_integral, indicator,
+                             dg, cache)
+  end
+end
+
+@inline function subcell_limiting_kernel!(du, u,
+                                          element, mesh::TreeMesh{2},
+                                          nonconservative_terms::False, equations,
+                                          volume_integral, indicator::IndicatorIDP,
+                                          dg::DGSEM, cache)
+  @unpack inverse_weights = dg.basis
+  @unpack volume_flux_dg, volume_flux_fv = volume_integral
+
+  # high-order DG fluxes
+  @unpack fhat1_threaded, fhat2_threaded = cache
+
+  fhat1 = fhat1_threaded[Threads.threadid()]
+  fhat2 = fhat2_threaded[Threads.threadid()]
+  calcflux_fhat!(fhat1, fhat2, u, mesh,
+      nonconservative_terms, equations, volume_flux_dg, dg, element, cache)
+
+  # low-order FV fluxes
+  @unpack fstar1_L_threaded, fstar1_R_threaded, fstar2_L_threaded, fstar2_R_threaded = cache
+
+  fstar1_L = fstar1_L_threaded[Threads.threadid()]
+  fstar2_L = fstar2_L_threaded[Threads.threadid()]
+  fstar1_R = fstar1_R_threaded[Threads.threadid()]
+  fstar2_R = fstar2_R_threaded[Threads.threadid()]
+  calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u, mesh,
+      nonconservative_terms, equations, volume_flux_fv, dg, element, cache)
+
+  # antidiffusive flux
+  calcflux_antidiffusive!(fhat1, fhat2, fstar1_L, fstar2_L, u, mesh,
+      nonconservative_terms, equations, indicator, dg, element, cache)
+
+  # Calculate volume integral contribution of low-order FV flux
+  for j in eachnode(dg), i in eachnode(dg)
+    for v in eachvariable(equations)
+      du[v, i, j, element] += inverse_weights[i] * (fstar1_L[v, i+1, j] - fstar1_R[v, i, j]) +
+                              inverse_weights[j] * (fstar2_L[v, i, j+1] - fstar2_R[v, i, j])
+
+    end
+  end
+
+  return nothing
+end
+
+
+#     calcflux_fhat!(fhat1, fhat2, u, mesh,
+#                    nonconservative_terms, equations, volume_flux_dg, dg, element, cache)
+#
+# Calculate the DG staggered volume fluxes `fhat` in subcell FV-form inside the element
+# (**without non-conservative terms**).
+#
+# See also `flux_differencing_kernel!`.
+#
+# # Arguments
+# - `fhat1::AbstractArray{<:Real, 3}`
+# - `fhat2::AbstractArray{<:Real, 3}`
```

Comment:
```suggestion
# Calculate the DG staggered volume fluxes `fhat` in subcell FV-form inside the element
# (**without non-conservative terms**).
#
# See also `flux_differencing_kernel!`.
```
This looks like the remnants of a docstring,  but since the code is right there, let's avoid the redundant information